src/transformStateWithClones.js

```diff
@@ -7,7 +7,46 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const actionLog = [];
+  let workState = Object.assign({}, state);
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(workState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        removeProperties(workState, action.keysToRemove);
+        break;
+
+      case 'clear':
+        workState = {};
+        break;
+
+      default:
+        return 'Error!';
+    }
+
+    actionLog.push(workState);
+    workState = Object.assign({}, workState);
```

Comment:
```suggestion
    actionLog.push({ ...workState });
```
also should work such.